kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java

```diff
@@ -84,6 +84,21 @@ public static void setResourceVersion(HasMetadata entity, String resourceVersion
     }
   }
 
+  /**
+   * Set namespace of a kubernetes resource if possible
+   *
+   * @param entity entity provided
+   * @param namespace the new namesapce
+   */
+  public static void setNamespace(HasMetadata entity, String namespace) {
+    if (entity != null) {
+      ObjectMeta metadata = entity.getMetadata();
+      if (metadata != null) {
+        metadata.setNamespace(namespace);
+      }
+    }
+  }
```

Comment:
I haven't checked the purpose of this util.
But was wondering if it might be interesting to initialize metadata instead of just ignoring if it's null:
```suggestion
  public static void setNamespace(HasMetadata entity, String namespace) {
    if (entity != null) {
      if (entity.getMetadata() == null) {
        entity.setMetadata(new ObjectMeta());
      }
      entity.getMetadata().setNamespace(namespace);
    }
  }
```

Reply:
It makes sense to me that in scope of this class, both methods should initialize the entity's metadata if necessary.
Maybe we can consider this as a future change.